src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -26,5 +28,44 @@ private static bool GetGlobalizationInvariantMode()
             }
             return invariantEnabled;
         }
+
+        private static bool LoadIcu()
+        {
+            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            {
+                return Interop.Globalization.LoadICU() != 0;
+            }
+
+#if TARGET_OSX
+            string extension = ".dylib";
+            bool versionAtEnd = false;
+#else
+            string extension = "so.";
+            bool versionAtEnd = true;
+#endif
+
+            // Append '.' to suffix since in Unix the version is separated by '.'
```

Comment:
Splitting the parsing and formatting over multiple places and trying too hard to make use Spans makes this hard to follow. 

You are about to load 10MB+ library. Allocating a few extra strings or going over the string a few more times does not matter.

I would make this as simple and as easy to follow as possible. The efficiency is secondary.

Reply:
Yeah I was doubtful of using span and didn’t know if was worth it because it is also a 1 path in an app lifetime plus only behind an opt-in feature. I was about to change it to strings but wanted to get a second opinion. Will simplify it to use strings, thanks for the thought.

Reply:
The optimization that tends to be more useful in cases like these is to have all handling of the the rare opt-in path in separate method. In this case, it would mean e.g. keep the regular load path inside `GetGlobalizationInvariantMode`, and change `LoadIcu` to `LoadAppLocalIcu(string icuVersionAndSuffix)`.

Reply:
Yeah, makes sense. Good idea, will do that.